os/arch/xtensa/src/esp32/Make.defs

```diff
@@ -124,13 +124,6 @@ ifeq ($(CONFIG_I2C),y)
 CHIP_CSRCS += esp32_i2c.c
 endif
 
-ifeq ($(CONFIG_FS_SMARTFS),y)
-CHIP_CSRCS += esp32_flash.c
```

Comment:
spiflash source file is built under the config of SMARTFS, while in esp_idf_port, it's depend on the WIFI support config, is it clear dependency for both external and os/arch  side ?

Reply:
this PR is to just remove legacy spi-flash driver, actually it does not used now, althou compiled.
then spi flash driver just only depends  WIFI config, not both.
it is better that make new PR to seperate spi flash driver from wifi support, i think.

Reply:
In my opinion, seperate spi flash driver from wifi support. is necessary.
  -- okay,  alreay seperate spi flash driver from wifi support, that is move spi flash driver 
to os/arch/xtensa/chip directory.